components/service/pocket/src/main/java/mozilla/components/service/pocket/PocketStoriesConfig.kt

```diff
@@ -6,22 +6,42 @@ package mozilla.components.service.pocket
 
 import mozilla.components.concept.fetch.Client
 import mozilla.components.support.base.worker.Frequency
+import java.util.UUID
 import java.util.concurrent.TimeUnit
 
 internal const val DEFAULT_REFRESH_INTERVAL = 4L
+internal const val DEFAULT_SPONSORED_STORIES_REFRESH_INTERVAL = 4L
 @Suppress("TopLevelPropertyNaming")
 internal val DEFAULT_REFRESH_TIMEUNIT = TimeUnit.HOURS
+@Suppress("TopLevelPropertyNaming")
+internal val DEFAULT_SPONSORED_STORIES_REFRESH_TIMEUNIT = TimeUnit.MINUTES
 
 /**
  * Indicating all details for how the pocket stories should be refreshed.
  *
  * @param client [Client] implementation used for downloading the Pocket stories.
- * @param frequency Optional - The interval at which to try and refresh items. Defaults to 1 hour.
+ * @param frequency Optional - The interval at which to try and refresh items. Defaults to 4 hours.
+ * @param profile Optional - The profile used for downloading sponsored Pocket stories.
+ * @param sponsoredStoriesRefreshFrequency Optional - The interval at which to try and refresh sponsored stories.
+ * Defaults to 30 minutes.
  */
 class PocketStoriesConfig(
     val client: Client,
     val frequency: Frequency = Frequency(
         DEFAULT_REFRESH_INTERVAL,
         DEFAULT_REFRESH_TIMEUNIT
+    ),
+    val profile: Profile? = null,
+    val sponsoredStoriesRefreshFrequency: Frequency = Frequency(
+        DEFAULT_SPONSORED_STORIES_REFRESH_INTERVAL,
+        DEFAULT_SPONSORED_STORIES_REFRESH_TIMEUNIT
     )
 )
+
+/**
+ * Sponsored stories configuration data.
+ *
+ * @param userId Unique identifier of the user which will be presented with sponsored stories.
+ * @param appId Unique identifier of the application using this feature.
+ */
+class Profile(val userId: UUID, val appId: String)
```

Comment:
How is the `userID` UUID handled and persisted? Does that happen in the consuming application? I saw https://github.com/mozilla-mobile/fenix/pull/25037/files#diff-989cb00682a18f94bf85f95b196445939a33ddf903cce43eb1a319cc13ae10e0R346 but wondering if that needs logic in A-C?

Reply:
I assume that it's send along with Glean telemetry later (in a separate PR), is that correct?

Reply:
Yes, I see both the `userId` (which can be reset when the user disables this feature) and the `appId` being handled in the client with AC providing just the functionality for "downloading" these stories.

Regarding the `userId` I was thinking about just generating one when the feature is enabled, persisting it in sharedPrefs, then deleting it when the feature is disabled.
It isn't needed in the Glean telemetry but by sending in in the download request it will impact the results we get.

Reply:
OK, makes sense!